src/textbooks/textbook-form/TextbookForm.test.jsx

```diff
@@ -0,0 +1,169 @@
+    </IntlProvider>
+  </AppProvider>,
+);
+
+describe('<TextbookForm />', () => {
+  beforeEach(async () => {
+    initializeMockApp({
+      authenticatedUser: {
+        userId: 3,
+        username: 'abc123',
+        administrator: true,
+        roles: [],
+      },
+    });
+
+    store = initializeStore();
+    axiosMock = new MockAdapter(getAuthenticatedHttpClient());
+  });
+
+  it('renders TextbooksForm component correctly', async () => {
+    const {
+      getByText, getByRole, getByPlaceholderText, getByTestId,
+    } = renderComponent();
+
+    await waitFor(() => {
+      expect(getByText(`${messages.tabTitleLabel.defaultMessage} *`)).toBeInTheDocument();
+      expect(getByPlaceholderText(messages.tabTitlePlaceholder.defaultMessage)).toBeInTheDocument();
+      expect(getByText(messages.tabTitleHelperText.defaultMessage)).toBeInTheDocument();
+
+      expect(getByText(`${messages.chapterTitleLabel.defaultMessage} *`)).toBeInTheDocument();
+      expect(getByPlaceholderText(
+        messages.chapterTitlePlaceholder.defaultMessage.replace('{value}', initialFormValuesMock.chapters.length),
+      )).toBeInTheDocument();
+      expect(getByText(messages.chapterTitleHelperText.defaultMessage)).toBeInTheDocument();
+
+      expect(getByText(`${messages.chapterUrlLabel.defaultMessage} *`)).toBeInTheDocument();
+      expect(getByPlaceholderText(messages.chapterUrlPlaceholder.defaultMessage)).toBeInTheDocument();
+      expect(getByText(messages.chapterUrlHelperText.defaultMessage)).toBeInTheDocument();
+
+      expect(getByTestId('chapter-upload-button')).toBeInTheDocument();
+      expect(getByTestId('chapter-delete-button')).toBeInTheDocument();
+
+      expect(getByRole('button', { name: messages.addChapterButton.defaultMessage }));
+      expect(getByRole('button', { name: messages.cancelButton.defaultMessage }));
+      expect(getByRole('button', { name: messages.saveButton.defaultMessage }));
+    });
+  });
+
+  it('calls onSubmit when the "Save" button is clicked with a valid form', async () => {
+    const { getByPlaceholderText, getByRole } = renderComponent();
+
+    const tabTitleInput = getByPlaceholderText(messages.tabTitlePlaceholder.defaultMessage);
+    const chapterInput = getByPlaceholderText(
+      messages.chapterTitlePlaceholder.defaultMessage.replace('{value}', initialFormValuesMock.chapters.length),
+    );
+    const urlInput = getByPlaceholderText(messages.chapterUrlPlaceholder.defaultMessage);
+
+    const formValues = {
+      tab_title: 'Tab title',
+      chapters: [
+        {
+          title: 'Chapter',
+          url: 'Url',
+        },
+      ],
+    };
+
+    userEvent.type(tabTitleInput, formValues.tab_title);
+    userEvent.type(chapterInput, formValues.chapters[0].title);
+    userEvent.type(urlInput, formValues.chapters[0].url);
+
+    await act(async () => {
+      userEvent.click(getByRole('button', { name: messages.saveButton.defaultMessage }));
+    });
+
+    await waitFor(() => {
```

Comment:
[nit]: This test will work successfully without `waitFor`. Let's delete it

Reply:
not working without `waitFor`